src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -1309,7 +1310,7 @@ private void WriteOpcContentTypes(ZipArchive package, HashSet<string> extensions
                     new XAttribute("Extension", "psmdcp"),
                     new XAttribute("ContentType", "application/vnd.openxmlformats-package.core-properties+xml"))
                     );
-            foreach (var extension in extensions)
+            foreach (var extension in extensions.ToImmutableSortedSet())
```

Comment:
instead of doing the sorting here it'd probably be easier to change the caller to pass in a SortedSet